scene/3d/lightmap_gi.cpp

```diff
@@ -419,7 +422,9 @@ int LightmapGI::_bsp_get_simplex_side(const Vector<Vector3> &p_points, const Loc
 //#define DEBUG_BSP
 
 int32_t LightmapGI::_compute_bsp_tree(const Vector<Vector3> &p_points, const LocalVector<Plane> &p_planes, LocalVector<int32_t> &planes_tested, const LocalVector<BSPSimplex> &p_simplices, const LocalVector<int32_t> &p_simplex_indices, LocalVector<BSPNode> &bsp_nodes) {
-	//if we reach here, it means there is more than one simplex
+	ERR_FAIL_COND_V(p_simplex_indices.size() == 0, -1);
+	ERR_FAIL_COND_V(p_simplex_indices.size() == 1, -(p_simplex_indices[0] + 1));
```

Comment:
Why does this return a different value? Errors shouldn't be relied upon value wise I'd say

Reply:
That is a fair question. If you call this function directly with only one simplex, it's an error that can't really be gracefully recovered from. But this function is also called recursively. And while it should never be called with only one simplex like that either, we can actually salvage the situation with a valid return value then.

Reply:
I'd say it should either return `-1` and throw an error, or recover without an error and have a valid return value, not return a valid value *and* error, the assumption would be that error means unusable (and should return a value that implies this when possible)

Reply:
OK